self-hosting/README.md

```diff
@@ -0,0 +1,329 @@
+
+## Installing Docker Prerequisites
+
+This Ideally depends on which operating system you are running. You need to first install these on your box before you can run the CHT docker containers.
+
+### Minimum Hardware/Software Requirements
+
+The specifications listed here are the minimum required to host the CHT containers. Depending on the scale of your operation these need to be adjusted to suite your need. 
+
+- 4 GiB RAM
+- 2 CPU/vCPU
+- 50 GB Hard Disk (SSD prefered)
+- SSL certificates ( To be able to use the CHT app on mobile)
+- Root Access to the host server
+
+### 64-bit Linux
+
+Depending on which distro you run, install the docker packages from [here](https://docs.docker.com/engine/install/). We have however, historically run most of our containers in Ubuntu.
+
+#### 64-bit Ubuntu
+
+- [Docker CE](https://docs.docker.com/engine/install/ubuntu/)  
+
+- [Docker-compose](https://docs.docker.com/compose/install/)
+
+### Windows 
+
+- [WSL2](https://docs.microsoft.com/en-us/windows/wsl/install-win10)
+
+- [Docker for Windows here](https://download.docker.com/win/stable/Docker%20for%20Windows%20Installer.exe)  and [here](https://docs.docker.com/docker-for-windows/install/)
+
+Note: If you have Hyper-V Capability, please ensure it is enabled in order to run Linux Containers on Windows. If you are running your Windows Server in cloud services, please ensure it is running on [bare-metal](https://en.wikipedia.org/wiki/Bare_machine). You will not be able to run Linux Containers in Windows if the previous comments are not adhered due to nested virtualization.
+
+If you do not have Hyper-V capability, but your server still supports virtualization, ensure that is enabled in your BiOS, and install the following package: [Docker Toolbox using VirtualBox](https://github.com/docker/toolbox/releases)
+
+### Mac OSX:
+
+- [Docker for Mac](https://download.docker.com/mac/stable/Docker.dmg)
+
+## Verify install
+
+To test that you installed docker and docker-compose correctly, make sure you can do a simple test run by showing their versions: Note,your version may be different.
+
+```bash
+
+sudo docker-compose --version
+docker-compose version 1.27.1, build 509cfb99
+
+sudo docker --version
+Docker version 19.03.12, build 48a66213fe
+
+## check that docker runs well on your server. 
+
+sudo docker run hello-world
+
+```
+
+## Installing with a compose file
+
+The CHT containers are installed using [docker compose](https://docs.docker.com/compose/reference/overview/) so that you can run multiple containers  as a single service.
+
+Start by choosing the location where you would like to save your compose configuration file.  Then create your docker compose file (`docker-compose.yml`) with contents as listed below.
```

Comment:
The `docker-compose.yml` file below is almost identical to [the one in this repo](https://github.com/medic/cht-infrastructure/blob/master/self-hosting/main/docker-compose.yml). Maybe we should update that file and replace this whole section with:

> Start by choosing the location where you would like to save your compose configuration file.  Then create your docker compose file (`docker-compose.yml`) by `cd`ing into the correct directory and running:
> ```
> curl -q -o docker-compose.yml https://raw.githubusercontent.com/medic/cht-infrastructure/master/self-hosting/main/docker-compose.yml
> ```

Either that or remove the `docker-compose.yml` file in this repo to reduce both confusion and the requirement to update both.